### PR TITLE
Update 'Understanding EF Services' for RC2

### PR DESCRIPTION
**Pull request by natemcmaster**
I update this guide to reflect the big changes we made in RC2 to isolate our internal services.

cc @ajcvickers @rowanmiller @divega

### CODE REVIEW

**Comment by rowanmiller** [comment]
Looks good to me, but would be good for @ajcvickers to review for technical accuracy.

File: docs/internals/services.rst [hunk 1]
```diff
@@ -15,6 +15,22 @@ which is implemented in EF using `Microsoft.Extensions.DependencyInjection
 This article covers some fundamentals principles for understanding how EF uses
 services and DI.
 
+Terms
+-----
+
+Service
+  A reusable component. In EF, a service identified by a .NET type (a class or interface).
```
Comment (ajcvickers) [inline docs/internals/services.rst]: a service _is_ identified
Also, all our services are identified by interfaces.

**Comment by ajcvickers** [comment]
Looks good. I might think about it a bit to see if a slightly different organization could make things clearer, but :shipit:

**Comment by natemcmaster** [comment]
How about this organization?

```
Terms
Service lifetimes
Category of EF services
Using EF in DI vs how EF uses DI
    Isolation of internal and external service provider
    AddDbContext and special cases
    EF's internal service provider (move content of service provider caching here)
    Configuring the internal service provider
```

Move "Required provider services" to the "writing a provider" doc.

**Comment by rowanmiller** [comment]
Lets :shipit:  as-is so that we can move forward with RC2 docs, and then improve the structure later

**Pull request merged**